src/loc/LocPrivateFileButton.tsx

```diff
@@ -98,13 +100,19 @@ export function LocPrivateFileButton() {
                     colors={ colorTheme.dialog }
                     onFileSelected={ handleSelectedFile }
                 />
+                { status === 'Hashing' &&
+                    <Row>
+                        <Spinner animation="border" size="sm" />
+                        &nbsp;Calculating Hash
+                    </Row>
+                }
                 { status === 'Uploading' &&
                     <Row>
                         <Spinner animation="border" size="sm" />
                         &nbsp;Uploading and encrypting
                     </Row>
                 }
-                { status !== 'Uploading' &&
+                { status !== 'Hashing' && status !== 'Uploading' &&
```

Comment:
You could also just test `uploadError` value (this requires the value to be reset on submit).